src/main/java/com/github/javachat/doublerange/DoubleRange.java

```diff
@@ -0,0 +1,385 @@
+package com.github.javachat.doublerange;
+
+import com.github.javachat.common.BoundType;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.text.NumberFormat;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.DoublePredicate;
+
+/**
+ * A range of double values.
+ * <p>
+ * This is meant as a drop-in replacement for
+ * <a href="http://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/Range.html">Range&lt;Double&gt;</a>,
+ * therefore the semantics of the Guava class apply to this class as well.
+ * <p>
+ * In regard to double specifics, this implementation follows Java semantics, this means:
+ * <p>
+ * <ul>
+ * <li>Any double lies between +Infinity and -Infinity</li>
+ * <li>NaN always lies in an infinite range</li>
+ * <li>NaN never lies in a range with discrete boundaries</li>
+ * </ul>
+ */
+public class DoubleRange {
+    /**
+     * The infinite range that spans every double value (but not {@link Double#NaN})
+     */
+    static final DoubleRange INFINITE_RANGE = new DoubleRange(Double.NEGATIVE_INFINITY, BoundType.OPEN,
+            Double.POSITIVE_INFINITY, BoundType.OPEN);
+
+    // Exception messages
+    static final String ILLEGAL_BOUND = "lower bound %f must be less than or equal " +
+            "to upper bound %.5f";
+    static final String ILLEGAL_OPEN_RANGE = "when specifying an open range, "
+            + "the lower bound must be strictly less than the upper bound";
+    static final String BOUNDARY_IS_NAN = "Both boundaries must be a number";
+    static final String NO_CONNECTION = "Cannot create intersection from two unconnected ranges (%s and %s)";
```

Comment:
To quote from Guava's [intersection(Range)](http://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/Range.html#intersection(com.google.common.collect.Range)) javadoc:

> Throws:
> IllegalArgumentException - if isConnected(connectedRange) is false